src/plugins/discover/public/application/apps/main/components/doc_table/doc_table_embeddable.tsx

```diff
@@ -21,33 +21,55 @@ export interface DocTableEmbeddableProps extends DocTableProps {
   totalHitCount: number;
 }
 
+const scrollTop = () => {
+  const scrollDiv = document.querySelector('.kbnDocTableWrapper') as HTMLElement;
+  scrollDiv.scrollTo(0, 0);
+};
+
 const DocTableWrapperMemoized = memo(DocTableWrapper);
 
 export const DocTableEmbeddable = (props: DocTableEmbeddableProps) => {
-  const pager = usePager({ totalItems: props.rows.length });
+  const {
+    currentPage,
+    pageSize,
+    totalPages,
+    startIndex,
+    hasNextPage,
+    changePage,
+    changePageSize,
+  } = usePager({
+    totalItems: props.rows.length,
+  });
 
-  const pageOfItems = useMemo(
-    () => props.rows.slice(pager.startIndex, pager.pageSize + pager.startIndex),
-    [pager.pageSize, pager.startIndex, props.rows]
-  );
+  const pageOfItems = useMemo(() => props.rows.slice(startIndex, pageSize + startIndex), [
+    pageSize,
+    startIndex,
+    props.rows,
+  ]);
 
-  const shouldShowLimitedResultsWarning = () =>
-    !pager.hasNextPage && props.rows.length < props.totalHitCount;
+  const onPageChange = useCallback(
+    (page: number) => {
+      scrollTop();
+      changePage(page);
+    },
+    [changePage]
+  );
 
-  const scrollTop = () => {
-    const scrollDiv = document.querySelector('.kbnDocTableWrapper') as HTMLElement;
-    scrollDiv.scrollTo(0, 0);
-  };
+  const onPageSizeChange = useCallback(
+    (size: number) => {
+      scrollTop();
+      changePageSize(size);
+    },
+    [changePageSize]
+  );
 
-  const onPageChange = (page: number) => {
-    scrollTop();
-    pager.onPageChange(page);
-  };
+  /**
+   * Go to the first page if filter was applied
+   */
+  useEffect(() => onPageChange(0), [onPageChange, props.rows.length]);
```

Comment:
I would suggest to change the page to the first, if data changes in a way, that the current page is no longer available. you know you could configure the dashboard to refresh very 5 seconds, are on the 2nd page, maybe the 10th page is no longer available, and this would automatically switch to the first page then. So if you are on the 10th page, and due to data change this page is no longer available, then I'd suggest to change the page. WDYT?

Reply:
Generally I am ok with that. On the other hand in this case when filter will be applied, the content of the current page will be changed anyway, however user will stay on it.